python/GangaLHCb/Lib/Applications/BenderBox.py

```diff
@@ -0,0 +1,520 @@
+        ## from GaudiExec 
+        'uploadedInput'    : GangaFileItem(defvalue=None, hidden=1, doc='This stores the input for the job which has been pre-uploaded so that it gets to the WN'),
+        'jobScriptArchive' : GangaFileItem(defvalue=None, hidden=1, copyable=0, doc='This file stores the uploaded scripts which are generated fron this app to run on the WN'),
+        'useGaudiRun'      : SimpleItem(defvalue=False, hidden = 1, doc='Should be False to call getWNPythonContents function'),
+        'extraOpts'        : SimpleItem(defvalue='', typelist=[str], hidden=1, doc='DO NOT USE IT; An additional string which is to be added to \'options\' when submitting the job'),
+        'options'          : GangaFileItem ( defvalue=[], sequence=1, hidden = 1,doc='List of files which contain the options to pass to gaudirun.py'),
+        'extraArgs'        : SimpleItem(defvalue=[], typelist=[list], sequence=1, hidden=1, doc=' Do NOT USE IT; Extra runtime arguments which are passed to the code running on the WN'),
+        # Prepared job object
+        'is_prepared'      : SimpleItem(defvalue=None, strict_sequence=0, visitable=1, copyable=1, hidden=0, typelist=[None, ShareDir], protected=0, comparable=1,
+                                        doc='Location of shared resources. Presence of this attribute implies the application has been prepared.'),
+        'hash'             : SimpleItem(defvalue=None, typelist=[None, str], hidden=1, doc='MD5 hash of the string representation of applications preparable attributes'),
+        })
+    _category      = 'applications'
+    _name          = 'BenderBox'
+    _exportmethods = ['prepare', 'unprepare' ]
+
+    def configure(self, masterjobconfig):	
+        self.options = self.tool.inputfiles()
+        return (None,None)
+
+    # =======================================================================
+    def getWNPythonContents(self):
+        """Return the wrapper script which is used to run Bender on the WN
+        """
+        from ..RTHandlers.GaudiExecRTHandlers import GaudiExecDiracRTHandler
+        data_file = GaudiExecDiracRTHandler.data_file
+        return self.tool.getWrapper(data_file)
+
+
+# =============================================================================
+## prepare Bender application
+#  @code
+#  j = Job ( application = prepareBender ( 'v30r1' ) ) 
+#  @endcode
+#  One can specify the directory explicitely:
+#  @code
+#  j = Job ( application = prepareBender ( 'v30r1' , path = '$HOME/mydir' ) ) 
+#  @endcode
+#  or use the temporary directory 
+#  @code
+#  j = Job ( application = prepareBender ( 'v30r1' , use_tmp = True  ) ) 
+#  @endcode
+#  One can also use the configuration parameters:
+#  @code
+#  j = Job ( application = prepareBender ( 'v30r1' , module = 'the_path/the_module.py' , params = { ...} ) ) 
+#  @endcode
+def prepareBender ( version  , path = '$HOME/cmtuser', use_tmp = False , **kwargs ) :
```

Comment:
> I think this is the Bender version of prepareGaudiExec. It checks out a clean version of Bender for you.

Ok this is why I left it to you guys, I'm showing my age already here as I remember the good old days when we use to just bind an application instance directly as the input parameter to a job rather than using what to me looks like a factory function. I wasn't even aware of this `prepareGaudiExec`. My question however I think still stands and applies equally to this function as well. Are they doing something that the prepare step, namely 
```python
Bender().prepare()
```
does not do?